sky/cli.py

```diff
@@ -1913,28 +1928,67 @@ def _down_or_stop_clusters(
         # Make sure the reserved clusters are explicitly specified without other
         # normal clusters and purge is True.
         if len(reserved_clusters) > 0:
+            if len(names) != 0:
+                names_str = ', '.join(map(repr, names))
+                raise click.UsageError(
+                    f'{operation} reserved cluster(s) '
+                    f'{reserved_clusters_str} with multiple other cluster(s) '
+                    f'{names_str} is not supported.\n'
+                    f'Please omit the reserved cluster(s) {reserved_clusters}.')
             if not down:
                 raise click.UsageError(
                     f'{operation} reserved cluster(s) '
                     f'{reserved_clusters_str} is not supported.')
             else:
-                click.secho(
-                    'WARNING: Tearing down a SkyPilot reserved cluster. If '
-                    'any related job had not finished, resources leakage '
-                    'could happen.',
-                    fg='yellow')
+                # TODO(zhwu): We can only have one reserved cluster (spot
+                # controller).
+                assert len(reserved_clusters) == 1, reserved_clusters
+                # spot_jobs will be empty when the spot cluster is not running.
+                cluster_name = reserved_clusters[0]
+                cluster_status, _ = backend_utils.refresh_cluster_status_handle(
+                    cluster_name)
+                if cluster_status is None:
+                    click.echo(
+                        'Managed spot controller has already been torn down.')
+                    return
+
+                cnt = 1
+                msg = (
+                    f'{colorama.Fore.YELLOW}WARNING: Tearing down the managed '
+                    f'spot controller ({cluster_status.value}). Please be '
+                    f'aware of the following:{colorama.Style.RESET_ALL}'
+                    f'\n {cnt}. All logs and status information of the spot '
+                    'jobs will be lost.')
+                cnt += 1
+                if cluster_status == global_user_state.ClusterStatus.INIT:
+                    msg += (
+                        f'\n {cnt}. Resource leakage may happen caused by '
+                        'spot jobs being submitted, and in-progress spot jobs.')
```

Comment:
nit: If there are pending/in-progress spot jobs, those resources will not be terminated and require manual cleanup.

Actually, why do we show this when the controller is INIT?

Reply:
Because it is possible that another `sky spot launch` is running in parallel, which will make the cluster status `INIT`, i.e. during a spot job is being submitted to the spot controller.